src/main/java/nextstep/subway/applicaion/ShortestDistancePathFinder.java

```diff
@@ -0,0 +1,84 @@
+package nextstep.subway.applicaion;
+
+import java.util.Collection;
+import java.util.Optional;
+import nextstep.subway.domain.PathFinder;
+import nextstep.subway.domain.Section;
+import nextstep.subway.domain.Station;
+import nextstep.subway.domain.vo.Path;
+import nextstep.subway.ui.BusinessException;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.WeightedMultigraph;
+
+public class ShortestDistancePathFinder implements PathFinder {
+
+  private final WeightedMultigraph<Station, PathWeightedEdge> graph;
+
+  public ShortestDistancePathFinder(final Collection<Section> sections) {
+    this.graph = buildGraph(sections);
+  }
+
+  @Override
+  public Optional<Path> find(final Station source, final Station target) {
+    verifyRequiredArguments(source, target);
+
+    final var path = new DijkstraShortestPath<>(graph).getPath(source, target);
+
+    if (path == null) {
+      return Optional.empty();
+    }
+
+    final int distance = path.getEdgeList().stream()
+        .mapToInt(PathWeightedEdge::getDistance)
+        .sum();
+
+    final int duration = path.getEdgeList().stream()
+        .mapToInt(PathWeightedEdge::getDuration)
+        .sum();
+
+    return Optional.of(path)
+        .map(it -> Path.from(it.getVertexList(), distance, duration));
+  }
+
+  @Override
+  public boolean isPathExists(Station source, Station target) {
+    final var path = new DijkstraShortestPath<>(graph).getPath(source, target);
```

Comment:
계속 만들지 않고 DijkstraShortestPath도 멤버 변수로 가질 수 있지 않을까요?
```suggestion
    final var path = dijkstraShortestPath.getPath(source, target);
```

Reply:
저도 그게 좋다고 생각하지만 그렇게되면 구간이 추가될 때 동기화하는 로직을 먼저 구현해야 될 것 같아요 🥲

Reply:
이 코멘트 다시 보니 제가 이해를 잘못했었군요!
말씀하신대로 graph가 아니라 path를 멤버변수로 가지고 있어도 로직 수행에 지장이 없네요
감사합니다 👍👍